packages/webchat/src/core/api.tsx

```diff
@@ -231,10 +196,5 @@ export default class WebchatApi {
       console.error('Conversation not found, starting a new one...')
       await this.createConversation()
     }
-
-    if (data.errorCode === 'BP_0401') {
-      this.updateAxiosConfig({ botId: this.botId, externalAuthToken: undefined })
-      console.error('External token expired or invalid. Removed from future requests')
-    }
   }
```

Comment:
When you configure the webchat, you can also set an "externalAuthToken" (a JWT token). Everytime you send a message, an event or some other requests, we validate that token and if it is valid, we add the property ".credentials" to the event when sending it to the event engine.

In essence, this is an authentication mechanism to compensate for the lack of security provided by the "visitorId". 

If the token is deemed invalid, we remove it from future requests to avoid parsing it each time

Reply:
We now have secure connection with the userId and userToken. Do we still need this feature?

Reply:
The externalAuthToken also provides more informations, since you can put anything in that payload (and can be assured that the content of `event.credentials` is safe & valid). Securing with a id/token is a good start, but it doesn't really replace this feature.

@EFF @slvnperron i'm not entirely sure how it's being used by clients (or if it is being used). Your thoughts?